hub/authorization.go

```diff
@@ -71,10 +74,28 @@ func authorize(r *http.Request, jwtKey []byte, publishAllowedOrigins []string) (
 // validateJWT validates that the provided JWT token is a valid Mercure token
 func validateJWT(encodedToken string, key []byte) (*claims, error) {
 	token, err := jwt.ParseWithClaims(encodedToken, &claims{}, func(token *jwt.Token) (interface{}, error) {
-		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
-			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		switch token.Method.(type) {
+			case *jwt.SigningMethodHMAC:
+				return key, nil
+			case *jwt.SigningMethodRSA:
+				block, _ := pem.Decode(key)
+
+				if block == nil {
+					return nil, errors.New("public key error")
```

Comment:
Shouldn't we return the error returned by `Decode` instead? It will be easier to debug.

Reply:
That's the thing: it does not return one:

```go
func Decode(data []byte) (p *Block, rest []byte) {
```

Reply:
Fine then!